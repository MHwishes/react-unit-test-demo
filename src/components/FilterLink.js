import {connect} from 'react-redux'
import {setVisibilityFilter} from '../actions'
import React, {Component} from "react";
import {lowerCase} from 'lodash';

class Link extends Component {

    render() {
        const {active, children, filter} = this.props;
        const disabled = active === filter;
        return (
            <button
                onClick={() => this.props.setVisibilityFilter(filter)}
                disabled={disabled}
                className={lowerCase(children)}
                style={{
                    marginLeft: '60px',
                    color: disabled ? '#36a1da' : 'black'
                }}
            >
                {children}
            </button>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        active: state.visibilityFilter
    }
};

const mapDispatchToProps = {
    setVisibilityFilter
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Link)
