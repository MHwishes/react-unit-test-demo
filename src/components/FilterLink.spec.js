import {mount, shallow} from 'enzyme';
import FilterLink from './FilterLink'
import configMockStore from 'redux-mock-store';
import React from 'react'

const mockStore = configMockStore();
import {visibilityFilters} from '../constant/constants'

const initialState = {
    todos: [
        {id: 0, text: "I have a meeting at 3PM", completed: true},
        {id: 1, text: "I go home at 6PM", completed: false}
    ],
    visibilityFilter: visibilityFilters.SHOW_ALL
};
describe('When user click footer button', () => {
    // todo: need to change
    it('should show all items when click `all` button', () => {
        const store = mockStore(initialState);
        const props = {
            active: 'SHOW_ALL',
            children: 'All',
            filter: 'SHOW_ALL'
        };
        const wrapper = mount(<FilterLink store={store} {...props}/>);
        wrapper.find('.all').simulate('click');
    });

    it('should show all items when click `completed` button', () => {
        const store = mockStore(initialState);
        const props = {
            active: 'SHOW_ALL',
            children: 'Completed',
            filter: 'SHOW_COMPLETED'
        };
        const wrapper = mount(<FilterLink store={store} {...props}/>);
        wrapper.find('.completed').simulate('click');
        expect(store.getActions()).toEqual(
            [{"payload": "SHOW_COMPLETED", "type": "SET_VISIBILITY_FILTER"}]
        );
    });


    it('should show all items when click `active` button', () => {
        const store = mockStore(initialState);
        const props = {
            active: 'SHOW_ALL',
            children: 'Active',
            filter: 'SHOW_ACTIVE'
        };
        const wrapper = mount(<FilterLink store={store} {...props}/>);
        wrapper.find('.active').simulate('click');
        expect(store.getActions()).toEqual(
            [{"payload": "SHOW_ACTIVE", "type": "SET_VISIBILITY_FILTER"}]
        );
    });
});
