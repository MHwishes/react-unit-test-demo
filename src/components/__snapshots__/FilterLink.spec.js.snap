// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When user click footer button should show all items when click \`active\` button 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "marginLeft": "60px",
    }
  }
/>
`;

exports[`When user click footer button should show all items when click \`all\` button 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "marginLeft": "60px",
    }
  }
/>
`;

exports[`When user click footer button should show all items when click \`completed\` button 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "marginLeft": "60px",
    }
  }
/>
`;

exports[`shallow render FilterLink component render FilterLink 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "marginLeft": "60px",
    }
  }
/>
`;
